File: Euclid/Game/Values/ValueManager.cs
using Euclid.Storage.Database.Access;
using Euclid.Util.Extensions;
using System.Collections.Generic;

namespace Euclid.Game
{
    public class ValueManager : ILoadable
    {
        #region Fields

        public static readonly ValueManager Instance = new ValueManager();
        private Dictionary<string, string> _clientValues;

        #endregion

        #region Properties

        private Dictionary<string, string> ClientValues
        {
            get { return _clientValues; }
        }

        #endregion

        #region Constructors

        public void Load()
        {
            SettingDao.GetSettings(out _clientValues);

            foreach (var kvp in GetDefaultValues())
            {
                if (!SettingDao.HasSetting(kvp.Key))
                {
                    SettingDao.SaveSetting(kvp.Key, kvp.Value);
                    _clientValues[kvp.Key] = kvp.Value;
                }
            }
        }

        /// <summary>
        /// Get default configuration values
        /// </summary>
        public Dictionary<string, string> GetDefaultValues()
        {
            var defaultValues = new Dictionary<string, string>();
            
            defaultValues["max.friends.normal"] = "300";
            defaultValues["max.friends.hc"] = "600";
            defaultValues["max.friends.vip"] = "1100";
            defaultValues["max.rooms.allowed"] = "100";
            defaultValues["max.rooms.allowed.subscribed"] = "200";
            defaultValues["timer.speech.bubble"] = "15";
            defaultValues["inventory.items.per.page"] = "500";
            defaultValues["catalogue.subscription.page"] = "63";
            defaultValues["club.gift.interval"] = "1";
            defaultValues["club.gift.interval.type"] = "MONTH";

            return defaultValues;
        }

        #endregion

        #region Public methods

        /// <summary>
        /// Get the integer value by key
        /// </summary>
        public int GetInt(string key)
        {
            if (ClientValues.TryGetValue(key, out string value))
                return value.IsNumeric() ? int.Parse(value) : 0;

            return 0;
        }

        /// <summary>
        /// Get the string value by key
        /// </summary>
        public string GetString(string key)
        {
            return ClientValues.TryGetValue(key, out string value) ? value : null;
        }

        #endregion
    }
}

File: euclid/src/main/java/euclid/game/values/ValueManager.java
package euclid.game.values;

import euclid.game.ILoadable;
import euclid.storage.database.access.SettingDao;

import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.Map;

public class ValueManager implements ILoadable {

    public static final ValueManager INSTANCE = new ValueManager();
    private Map<String, String> clientValues;

    private ValueManager() {
    }

    public static ValueManager getInstance() {
        return INSTANCE;
    }

    @Override
    public void load() {
        clientValues = SettingDao.getSettings();

        for (Map.Entry<String, String> entry : getDefaultValues().entrySet()) {
            if (!SettingDao.hasSetting(entry.getKey())) {
                SettingDao.saveSetting(entry.getKey(), entry.getValue());
                clientValues.put(entry.getKey(), entry.getValue());
            }
        }
    }

    public Map<String, String> getDefaultValues() {
        Map<String, String> defaults = new LinkedHashMap<>();
        defaults.put("max.friends.normal", "300");
        defaults.put("max.friends.hc", "600");
        defaults.put("max.friends.vip", "1100");
        defaults.put("max.rooms.allowed", "100");
        defaults.put("max.rooms.allowed.subscribed", "200");
        defaults.put("timer.speech.bubble", "15");
        defaults.put("inventory.items.per.page", "500");
        defaults.put("catalogue.subscription.page", "63");
        defaults.put("club.gift.interval", "1");
        defaults.put("club.gift.interval.type", "MONTH");
        return defaults;
    }

    public int getInt(String key) {
        String value = clientValues.get(key);
        if (value != null) {
            try {
                return Integer.parseInt(value);
            } catch (NumberFormatException e) {
                return 0;
            }
        }
        return 0;
    }

    public String getString(String key) {
        return clientValues.get(key);
    }
}
